packages/xrpl/test/setupClient.ts

```diff
@@ -1,64 +1,92 @@
-/* eslint-disable no-param-reassign -- Necessary for test setup */
-/* eslint-disable @typescript-eslint/explicit-module-boundary-types -- Necessary for test setup */
 import { Client, BroadcastClient } from 'xrpl-local'
 
-import createMockRippled from './createMockRippled'
-import { getFreePort } from './testUtils'
+import createMockRippled, {
+  type MockedWebSocketServer,
+} from './createMockRippled'
+import { destroyServer, getFreePort } from './testUtils'
+
+export interface XrplTestContext {
+  client: Client | BroadcastClient
+  _mockedServerPort?: number
+  mockRippled?: MockedWebSocketServer
+  mocks?: MockedWebSocketServer[]
+  servers?: number[]
+}
 
 async function setupMockRippledConnection(
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any -- Typing is too complicated
-  testcase: any,
   port: number,
-): Promise<void> {
-  return new Promise<void>((resolve, reject) => {
-    testcase.mockRippled = createMockRippled(port)
-    testcase._mockedServerPort = port
-    testcase.client = new Client(`ws://localhost:${port}`)
-    testcase.client.connect().then(resolve).catch(reject)
+): Promise<XrplTestContext> {
+  const context: XrplTestContext = {
+    mockRippled: createMockRippled(port),
+    _mockedServerPort: port,
+    client: new Client(`ws://localhost:${port}`),
+    servers: [port],
+  }
+
+  context.client.on('error', () => {
+    // We must have an error listener attached for reconnect errors
   })
+
+  return context.client.connect().then(() => context)
 }
 
 async function setupMockRippledConnectionForBroadcast(
```

Comment:
Isn't this no longer needed, now that the broadcast client tests have been deleted?

Reply:
Oh they were deleted? When?

Reply:
Added to follow up